uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -783,7 +784,13 @@ define([
      * @param {String} time - start time in HH:mm format
      */
     function updateEndTimeInputMinTime (time) {
-      var timeToMin = getMomentDateWithGivenTime(time)
+      var timeToMin;
+      var base = sharedSettings.timeBaseInMinutes * 60;
+
+      time = moment.unix(
```

Comment:
I think you have the potential to put 

```
moment.unix(
  Math.ceil(moment(time, timeFormat).unix() / base) * base)
    .format(timeFormat);
```

Inside a function that explains all that magic that is going on here and which can be reused in https://github.com/compucorp/civihr/pull/2856/files#diff-06deb86aea35fed830b02774a44a2e7bR602

Reply:
Done. The one problem is that it cannot really be **reused** because one is angular, the other is not. So I ended up creating two funcs.